docs/using-react-redux/static-types.md

```diff
@@ -0,0 +1,106 @@
+---
+id: static-types
+title: Type safety and React-Redux
+hide_title: true
+sidebar_label: Type safety and React-Redux
+---
+
+# Type safety and React-Redux
+
+The [`react-redux` type definitions](https://npm.im/@types/react-redux) export some helpers to make it easier to write typesafe interfaces between your redux store and your React components.
+
+## Typing the useSelector hook
+
+If you manually type your selector functions, `useSelector` will return the same type as your selector
+
+```ts
+// selectIsOn always returns a boolean, so isOn is typed as a boolean
+const isOn = useSelector(selectIsOn)
+```
+
+Otherwise `useSelector` requires you to manually type the state argument every time you use it.
+
+```ts
+const isOn = useSelector((state: MyStateType) => state.isOn)
+```
+
+If you'd like, you can define a typed `useSelect` hook using a helper type
+
+```ts
+// reducer.ts
+import { useSelector, TypedUseSelectorHook } from 'react-redux'
+
+interface MyStateType {
+  isOn: boolean
+}
+
+export const useTypedSelector: TypedUseSelectorHook<MyStateType> = useSelector
+
+// my-component.tsx
+import { useTypedSelector } from './reducer.ts'
+
+const isOn = useSelector(state => state.isOn)
+```
+
+## Typing the `connect` higher order component
+
+React-redux exposes a helper type, `ConnectedProps`, that can extract the return type of mapStateToProps and mapDispatchToProps.
```

Comment:
I'd like some explanatory stuff here about how `connect` is a HOC, and there's three separate sources of props: parent component, `mapState`, and `mapDispatch`.  

We should show a couple examples of "manually" typing `connect` and the plain component before introducing the `ConnectedProps<T>` technique.

Then again, I could see an argument for doing those _after_, or maybe even skipping them if we want to just highlight this technique.

Reply:
Tried flipping it around so that the example of manually typing connect comes first. Not sure what meaningful variations on this that I should cover.